### PR TITLE
Fix(eos_cli_config_gen): Fix the invalid configuration of vpn-route in export direction for router bgp vrf

### LINKED ISSUE #4731: Bug(eos_cli_config_gen): route-target export should have vrf-route instead of vpn-route for router bgp
### Issue Summary

Router BGP VRF configuration for route-target should have vrf-route in one direction(export) and vpn-route in the other direction(import) but right now bot have vpn-route. Hence it render the below configuration.
`route-target export vpn-ipv6 rcf RT_IMPORT_AF_RCF() vpn-route filter-rcf RT_IMPORT_AF_RCF_FILTER()`.

I should render below configuration.
`route-target export vpn-ipv6 rcf RT_IMPORT_AF_RCF() vrf-route filter-rcf RT_IMPORT_AF_RCF_FILTER()`

https://github.com/aristanetworks/avd/pull/4708#discussion_r1842767705

### Which component(s) of AVD impacted

eos_cli_config_gen

### How do you run AVD ?

Ansible CLI (with virtual-env or native python)

### Steps to reproduce

```shell
Use below input:

router_bgp:
  vrfs:
    - name: Test1
      route_targets:
        export:
          - address_family: evpn
            route_targets:
              - "1:30001"
            rcf: RT_EXPORT_AF_RCF()
```

### Relevant log output

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Fix the vrf-route for route-target export in router bgp vrf.

Router BGP VRF configuration for route-target should have vrf-route in one direction(export) and vpn-route in the other direction(import) but right now both have vpn-route. 

Hence it renders the below configuration.
route-target export vpn-ipv6 rcf RT_IMPORT_AF_RCF() vpn-route filter-rcf RT_IMPORT_AF_RCF_FILTER().

Expected configuration
route-target export vpn-ipv6 rcf RT_IMPORT_AF_RCF() vrf-route filter-rcf RT_IMPORT_AF_RCF_FILTER()

## Related Issue(s)

Fixes #4731 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Updated the j2 template to render the vrf-route instead of vpn-route for export direction in router bgp vrf.

## How to test
Run molecule and deploy the config on lab.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2 [hunk 1]
```diff
@@ -2678,7 +2678,7 @@ router bgp {{ router_bgp.as }}
 {%                 if address_family.address_family in ['evpn', 'vpn-ipv4', 'vpn-ipv6'] %}
 {%                     if address_family.rcf is arista.avd.defined %}
 {%                         if address_family.vpn_route_filter_rcf is arista.avd.defined and address_family.address_family in ['vpn-ipv4', 'vpn-ipv6'] %}
-      route-target export {{ address_family.address_family }} rcf {{ address_family.rcf }} vpn-route filter-rcf {{ address_family.vpn_route_filter_rcf }}
+      route-target export {{ address_family.address_family }} rcf {{ address_family.rcf }} vrf-route filter-rcf {{ address_family.vpn_route_filter_rcf }}
```
Comment (Vibhu-gslab) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2]: Attribute name should change to `address_family.vrf_route_filter_rcf` please change in schema as well, also change the description in schema from VPN to VRF (if needed).

Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2]: updated

**Review by Vibhu-gslab — APPROVED** [review]
LGTM!

**Review by laxmikantchintakindi — APPROVED** [review]
LGTM.

**Review by alexeygorbunov — APPROVED** [review]
LGTM

**Pull request merged**